src/ui/public/index_patterns/_calculate_indices.js

```diff
@@ -60,19 +59,27 @@ define(function (require) {
     }
 
     function sortIndexStats(indices, timeFieldName, sortDirection) {
-      if (!sortDirection) return _.keys(indices);
+      const desc = sortDirection === 'desc';
+      const leader = desc ? 'max' : 'min';
 
-      // FIXME: Once https://github.com/elastic/elasticsearch/issues/14404 is closed
-      // this should be sorting based on the sortable value of a field.
-      const edgeKey = sortDirection === 'desc' ? 'max_value' : 'min_value';
+      let indexDetails = _(indices).map((stats, index) => {
+        const field = stats.fields[timeFieldName];
 
-      return _(indices)
-      .map((stats, index) => (
-        { index, edge: stats.fields[timeFieldName][edgeKey] }
-      ))
-      .sortByOrder(['edge'], [sortDirection])
-      .pluck('index')
-      .value();
+        // TODO: remove when we get to es 2.2, see elastic/elasticsearch#14404
+        let min = field.min_value;
+        if (typeof min === 'string') min = moment(min).valueOf();
```

Comment:
After we upgrade to es 2.2, elasticsearch will always pass back unix timestamps. Until then times are passed back in whatever format they are stored in, and our only option is to attempt to parse that with `moment()`